jarviscli/plugins/OpenWebsite.py

```diff
@@ -0,0 +1,87 @@
+import webbrowser
+import os
+from plugin import plugin, require
+
+FILE_PATH = os.path.abspath(os.path.dirname(__file__))
+
+
+@require(network=True)
+@plugin("open website")
+class OpenWebsite:
+    """
+    This plugin will open a website using some params.
+
+    The user can open a simple website giving a complete link or
+    inputting the name of the website like the examples:
+
+    > open website www.google.com
+    > open website github
+    > open website github username
+
+    You can found a csv with the saved websites on:
+    Jarvis/jarviscli/data/website.csv
+    """
+    def __call__(self, jarvis, link):
+        inputs = link.split(' ', 1)
+        self.main_link = inputs[0]
+        self.complement = False
+        if len(inputs) > 1:
+            self.complement = inputs[1]
+
+        if self.has_on_saved_links():
+            webbrowser.open(self.main_link)
+        elif self.verify_link():
+            webbrowser.open(self.main_link)
+        else:
+            print("Sorry, I can't open this link.")
+
+    def has_on_saved_links(self):
+        websites_csv = \
+            open(os.path.join(FILE_PATH, "../data/websites.csv"), 'r')
+        for website in websites_csv:
+            information = website.split(',')
+            if self.main_link == information[0]:
+                if self.complement:
+                    if len(information) > 2:
+                        self.main_link = \
+                            information[1] + information[2] + self.complement
+                    else:
+                        self.main_link = information[1] + self.complement
+                else:
+                    self.main_link = information[1]
+                return True
+
+        return False
+
+    def verify_link(self):
+        if ((self.main_link[:8] != "https://" and
+             self.main_link[:7] != "http://" and
+             self.main_link[:3] != "www") or
+                ("com" not in self.main_link)):
```

Comment:
There are lots of other top-level domains besides '.com'...

https://en.wikipedia.org/wiki/List_of_Internet_top-level_domains

Instead of this check I would just resolve the domain. With Python e.g.:

```
socket.getaddrinfo("example.org", 80)
```
https://docs.python.org/3/library/socket.html#socket.getaddrinfo

And if this does not throw a socket.gaierror domain should be ok ;).

This module is also always usfull when dealing with urls:

https://docs.python.org/3/library/urllib.parse.html